dvc/dependency/param.py

```diff
@@ -125,9 +125,13 @@ def workspace_status(self):
             elif param not in info:
                 st = "new"
             elif actual[param] != info[param]:
-                st = "modified"
+                if isinstance(actual[param], tuple) and actual[param] == tuple(
+                    info[param]
+                ):
```

Comment:
Actually thinking about it some more, this is probably still wrong. If we have something like `info[param] = {1: 'foo', 2: 'bar'}` this will end up matching `actual[param] = tuple(1, 2)` since `tuple(dict)` will give you a tuple of the dict keys. (The same problem occurs if you do the original PR comparison with `len()` and then zip the iterators to compare each item in sequence))

I think it needs to be the other way around, where we convert the tuple into list like
```
if isinstance(actual[param], tuple) and list(actual[param]) == info[param]
```

Reply:
You are right. 

I was also thinking on it and had:

```
isinstance(actual[param], tuple) 
and isisntance(info[param]), list) 
and actual[param] == tuple(info[param])
```

But what you suggest is simpler